test/testuninitvar.cpp

```diff
@@ -6897,7 +6897,7 @@ class TestUninitVar : public TestFixture {
                         "    int a = ab.a;\n"
                         "    int b = ab.b;\n"
                         "}");
-        ASSERT_EQUALS("[test.cpp:5]: (error) Uninitialized variable: ab.b\n", errout.str());
+        TODO_ASSERT_EQUALS("[test.cpp:5]: (error) Uninitialized variable: ab.b\n", "", errout.str());
```

Comment:
Does the second forward overwrite values that were already set? Seems like MemberExpressionAnalyzer is doing the right thing here.

Reply:
The valueflow is correct for this case, the `.` in `ab.b` is a `Uninit`. For some reason, we dont warn in the checker. I am not sure why.

Reply:
Actually, its a valueflow issue, we are too aggressive at forwarding to the parent `=`.